### PR TITLE
Enable the JsonConvertibleParamConverter to convert deep objects

### PR DESCRIPTION
**Pull request by rjkip**

### CODE REVIEW

File: src/Request/JsonConvertibleParamConverter.php [hunk 1]
```diff
@@ -93,23 +90,27 @@ public function supports(ParamConverter $configuration)
...
     {
-        $snakeCase = '';
-
-        $len = strlen($camelCase);
-        for ($i = 0; $i < $len; $i++) {
-            if (ctype_upper($camelCase[$i])) {
-                $snakeCase .= '_'.strtolower($camelCase[$i]);
-            } else {
-                $snakeCase .= strtolower($camelCase[$i]);
+        $camelCasedObject = (object) [];
```
Comment (DRvanR) [inline src/Request/JsonConvertibleParamConverter.php]: just create a `\stdClass` then, as this leads to the same but in a less obvious way.

File: src/Request/JsonConvertibleParamConverter.php [hunk 2]
```diff
@@ -93,23 +90,27 @@ public function supports(ParamConverter $configuration)
...
+
+        foreach ($object as $property => $value) {
+            $camelCasedKey = is_string($property)
+                ? lcfirst(str_replace(' ', '', ucwords(str_replace('_', ' ', $property))))
+                : $property;
```
Comment (DRvanR) [inline src/Request/JsonConvertibleParamConverter.php]: wut? the key (propertyname) may not be a string? Or am I missing something?

**Comment by DRvanR** [comment]
Not a fan of this change, hides a lot of magic. What is the usecase?

File: src/Request/JsonConvertibleParamConverter.php (hunk 2 above)
Comment (rjkip) [inline src/Request/JsonConvertibleParamConverter.php]: Hmm, yeah, integer property names make no sense. I'll remove the string check.

**Comment by rjkip** [comment]
Use case would be the registration API parameter that would contain [the request and response objects](https://wiki.surfnet.nl/display/stepup/API+Design#APIDesign-Register) (as discussed yesterday), but this will not work as we lose type information of the request and response objects, which would become stdClasses. Or we'd have to move all request and response properties onto the registration object.

I'd rather keep the API the way it is, much simpler, and doesn't need the magic in this PR (which was not for review :wink:).

**Comment by DRvanR** [comment]
>I'd rather keep the API the way it is, much simpler, and doesn't need the magic in this PR
I don't see any simplicity or complexity gains either way tbh. The magic in this PR is not needed anyhow, rather than trying to generalize an existing solution for a specific problem any further to also solve the problem of deserialisation of nested objects a separate ParamConverter specifically for these (request & signing) request and response objects to me makes more sense.

That said, I'm not against the current API, I would prefer having nested objects (to me shows explicit coherence between the two rather than implicit through naming) but hardly think that's worth a fuss tbph. 

>(which was not for review :wink:).
:flushed: meh, just checked all repo's I've received an email from, skimming for a review tag (which also was removed here) :tongue:

**Pull request closed without merging**